core/trino-main/src/main/java/io/trino/cost/TableScanStatsRule.java

```diff
@@ -105,4 +107,23 @@ else if (type instanceof FixedWidthType) {
         });
         return result.build();
     }
+
+    private static double getNullsFraction(ColumnStatistics columnStatistics, Estimate rowCount)
+    {
+        if (columnStatistics.getNullsFraction().isUnknown() && !columnStatistics.getDistinctValuesCount().isUnknown()) {
```

Comment:
reverse iff condition, so that:
```
if (!columnStatistics.getNullsFraction().isUnknown() || columnStatistics.getDistinctValuesCount().isUnknown()) {
  columnStatistics.getNullsFraction().getValue();
}
```

you probably also need to check if `rowCount` is known